core/node/node_framework/src/implementations/layers/da_clients/eigen.rs

```diff
@@ -28,19 +39,42 @@ pub struct Output {
 
 #[async_trait::async_trait]
 impl WiringLayer for EigenWiringLayer {
-    type Input = ();
+    type Input = Input;
     type Output = Output;
 
     fn layer_name(&self) -> &'static str {
         "eigen_client_layer"
     }
 
-    async fn wire(self, _input: Self::Input) -> Result<Self::Output, WiringError> {
-        let client: Box<dyn DataAvailabilityClient> =
-            Box::new(EigenClient::new(self.config, self.secrets).await?);
+    async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
+        let master_pool = input.master_pool.get().await?;
+        let get_blob_from_db = GetBlobFromDB { pool: master_pool };
+        let client: Box<dyn DataAvailabilityClient> = Box::new(
+            EigenClient::new(self.config, self.secrets, Box::new(get_blob_from_db)).await?,
+        );
 
         Ok(Self::Output {
             client: DAClientResource(client),
         })
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct GetBlobFromDB {
+    pool: ConnectionPool<Core>,
+}
+
+#[async_trait::async_trait]
+impl GetBlobData for GetBlobFromDB {
+    async fn call(&self, input: &'_ str) -> anyhow::Result<Option<Vec<u8>>> {
```

Comment:
This lifetime can be skipped, I *think* the only reason it's not flagged by Clippy is that it's inside the `async_trait` proc macro.

Reply:
fixed here https://github.com/lambdaclass/zksync-era/pull/375